test/NuGet.Core.Tests/NuGet.ProjectModel.Test/CacheFileFormatTests.cs

```diff
@@ -3,57 +3,161 @@
 using System.Diagnostics;
 using System.IO;
 using System.Text;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using NuGet.Common;
 using NuGet.Frameworks;
+using NuGet.Test.Utility;
 using Xunit;
 
 namespace NuGet.ProjectModel.Test
 {
     public class CacheFileFormatTests
     {
-        [Theory]
-        [InlineData("1", "LhkXQGGI+FQMy9dhLYjG5sWcHX3z/copzi4hjjBiY3Fotv0i7zQCikMZQ+rOKJ03gtx0hoHwIx5oKkM7sVHu7g==", "true", true)]
-        [InlineData("2", "LhkXQGGI+FQMy9dhLYjG5sWcHX3z/copzi4hjjBiY3Fotv0i7zQCikMZQ+rOKJ03gtx0hoHwIx5oKkM7sVHu7g==", "true", false)]
-        public void CacheFileFormat_CacheFileReadCorrectly(string v, string dgSpecHash, string success, bool expectedValid)
+        [Fact]
+        public void CacheFileFormat_CacheFileReadCorrectlyVersion1()
         {
-            var cacheTemplate = @"{{
-  ""version"": {0},
-  ""dgSpecHash"": ""{1}"",
-  ""success"": {2}
+            var logger = new TestLogger();
+
+            var contents = $@"{{
+  ""version"": ""1"",
+  ""dgSpecHash"": ""LhkXQGGI+FQMy9dhLYjG5sWcHX3z/copzi4hjjBiY3Fotv0i7zQCikMZQ+rOKJ03gtx0hoHwIx5oKkM7sVHu7g=="",
+  ""success"": true,
 }}";
             CacheFile cacheFile = null;
-            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(string.Format(cacheTemplate, v, dgSpecHash, success))))
+            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(contents)))
             {
-                cacheFile = CacheFileFormat.Read(stream, NullLogger.Instance, "emptyPath");
+                cacheFile = CacheFileFormat.Read(stream, logger, "emptyPath");
+            }
+            Assert.False(cacheFile.IsValid);
+            Assert.True(cacheFile.Success);
+            Assert.Equal("LhkXQGGI+FQMy9dhLYjG5sWcHX3z/copzi4hjjBiY3Fotv0i7zQCikMZQ+rOKJ03gtx0hoHwIx5oKkM7sVHu7g==", cacheFile.DgSpecHash);
+            Assert.Equal(1, cacheFile.Version);
+
+            Assert.Equal(0, logger.Errors);
+            Assert.Equal(0, logger.Warnings);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CacheFileFormat_CacheFileReadCorrectly(bool haveMissingFile)
+        {
+            using (var workingDir = TestDirectory.Create())
+            {
+                var logger = new TestLogger();
+
+                var projectFullPath = Path.Combine(Path.GetTempPath(), "EA11D9B8013142A6B40A81FD90F57EAA");
+                var dgSpecHash = "LhkXQGGI+FQMy9dhLYjG5sWcHX3z/copzi4hjjBiY3Fotv0i7zQCikMZQ+rOKJ03gtx0hoHwIx5oKkM7sVHu7g==";
+                var success = "true";
+                
+                var file1 = Path.Combine(workingDir, "7A329DF71DDD41F689C9AD876DDF79F6");
+                var file2 = Path.Combine(workingDir, "C16089965CF84822A71D07580B29AF0E");
```

Comment:
Can you reuse `workingDir` instead of writing to `Path.GetTempPath()`?  `TestDirectory` will delete these files on dispose.  We've had disk space issues with tests creating files without cleaning them up.

Reply:
Done